operators/src/adapters/raster_subquery/raster_subquery_reprojection.rs

```diff
@@ -144,6 +116,98 @@ where
     }
 }
 
+fn projected_coordinate_grid_parallel(
+    pool: &ThreadPool,
+    tile_info: TileInformation,
+    out_srs: SpatialReference,
+    in_srs: SpatialReference,
+    valid_out_area: &SpatialPartition2D,
+) -> Result<Grid2D<Option<Coordinate2D>>> {
+    const OVERPROVISIONING: usize = 1; // split the work into multiple chunks per thread.
+    const MAX_PARALELLISM: usize = 8; // max task / work split
+
+    let start = std::time::Instant::now();
+
+    let res = pool.install(|| {
+        // get all pixel idxs and coordinates.
+        debug!(
+            "projected_coordinate_grid_parallel {:?}",
+            &tile_info.global_tile_position
+        );
+
+        let mut coord_grid: Grid2D<Option<Coordinate2D>> =
+            Grid2D::new_filled(tile_info.tile_size_in_pixels, None, None);
+
+        let tile_geo_transform = tile_info.tile_geo_transform();
+
+        let parallelism = pool.current_num_threads().max(MAX_PARALELLISM); // TODO: figure out how to enhance this
+        let par_chunk_split =
+            (tile_info.tile_size_in_pixels.axis_size_y() / (parallelism * OVERPROVISIONING)).max(1); // don't go below one line. Thats more complicated then needed
+        let par_chunk_size = tile_info.tile_size_in_pixels.axis_size_x() * par_chunk_split;
+        debug!(
+            "parallelism: threads={} par_chunk_split={} par_chunk_size={}",
+            pool.current_num_threads(),
+            par_chunk_split,
+            par_chunk_size
+        );
+
+        let axis_size_x = tile_info.tile_size_in_pixels.axis_size_x();
+
+        let res = coord_grid
+            .data
+            .par_chunks_mut(par_chunk_size)
+            .enumerate()
+            .try_for_each(|(chunk_idx, opt_coord_slice)| {
+                let chunk_start_y = chunk_idx * par_chunk_split;
+                let chunk_len = opt_coord_slice.len();
+                let out_coords = (0..chunk_len)
+                    .map(|lin_idx| {
+                        let x_idx = lin_idx % axis_size_x;
+                        let y_idx = lin_idx / axis_size_x + chunk_start_y;
+                        let grid_idx = GridIdx2D::from([y_idx as isize, x_idx as isize]);
+                        tile_geo_transform.grid_idx_to_upper_left_coordinate_2d(grid_idx)
+                    })
+                    .collect::<Vec<Coordinate2D>>();
+
+                let chunk_bounds =
+                    SpatialPartition2D::new_unchecked(out_coords[0], out_coords[chunk_len - 1]);
+
+                let proj = CoordinateProjector::from_known_srs(out_srs, in_srs)?;
```

Comment:
Is it necessary to call this per chunk? Couldn't this be initialized beforehand?

Reply:
i do not know but will test it. i thought proj objects are not threadsafe / send /sync.